### PR TITLE
OLP-39 load data from split pbfs

### PR DESCRIPTION
**Pull request by zandemax**

### CODE REVIEW

File: openlineplanner-backend/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ anyhow = "1.0.69"
 config = "0.13.3"
 fern = { version = "0.6.1", features = ["colored"] }
 geo = { version = "0.23.1", features = ["use-serde"] }
-geojson = { version = "0.24.0", features = ["geo-types"] }
+geojson = { git = "https://github.com/zandemax/geojson", features = ["geo-types"] }
```
Comment (TheNewCivilian) [inline openlineplanner-backend/Cargo.toml:15]: Why is this needed? Maybe add a comment.

Comment (TheNewCivilian) [reply openlineplanner-backend/Cargo.toml:15]: I think maintaining this aside is not a good idea. Is there an alternative way to fix this? i.g. manipulating the data to conform with the parser or move the parser to our project? If that cannot be achieved and it still is a project requirement, it should be moved to a private xatellite repo.

File: openlineplanner-backend/src/main.rs [hunk 2]
```diff
@@ -100,44 +96,33 @@ async fn main() -> std::io::Result<()> {
...
-                origin
-                    .as_bytes()
-                    .ends_with(b".openlineplanner.xatellite.space")
-            })
-            .allowed_origin_fn(|origin, _req_head| origin.as_bytes().ends_with(b"localhost:3000"))
-            .allowed_methods(vec!["GET", "POST", "DELETE"])
-            .allowed_headers(vec![http::header::AUTHORIZATION, http::header::ACCEPT])
-            .allowed_header(http::header::CONTENT_TYPE)
-            .max_age(3600);
+            .allow_any_origin()
```
Comment (TheNewCivilian) [inline openlineplanner-backend/src/main.rs]: Lets make this more restrictive again.. but fixing it in a separate new issue.

File: openlineplanner-backend/src/main.rs [hunk 3]
```diff
@@ -100,44 +96,33 @@ async fn main() -> std::io::Result<()> {
     #[rustfmt::skip]
     let config = Config::builder()
         .set_default("cache.dir", "./cache/").unwrap()
-        .set_default("data.dir", "./pbf/").unwrap()
+        .set_default("data.dir", "./data/").unwrap()
```
Comment (TheNewCivilian) [inline openlineplanner-backend/src/main.rs:99]: Nice!

File: openlineplanner/src/components/MapStationPopup.vue [hunk 4]
```diff
@@ -189,7 +189,7 @@ export default {
   // align-items: center;
   justify-content: center;
   flex-direction: column;
-  z-index: 4;
+  z-index: 10;
```
Comment (TheNewCivilian) [inline openlineplanner/src/components/MapStationPopup.vue]: Did it fix any issue? I know there are still some layover bugs occurring.. lets fix them soon

File: tools/regionalextracts/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+[package]
```
Comment (TheNewCivilian) [inline tools/regionalextracts/Cargo.toml:1]: Nice project!

File: tools/regionalextracts/src/admin_area.rs [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+struct Context {
+    admin_level: u16,
+    area_id: u64,
+}
+
+pub fn render_ovp_query_template(admin_level: u16, area_id: u64) -> Result<String, Error> {
+    let mut tt = TinyTemplate::new();
+    tt.add_template("query", OVP_QUERY_TEMPLATE)?;
+
+    let area_id = area_id + 3600000000;
```
Comment (TheNewCivilian) [inline tools/regionalextracts/src/admin_area.rs:101]: A comment would be good, why this is needed.

File: tools/regionalextracts/src/main.rs [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+enum Commands {
+    Split {
+        levels: Option<Vec<u16>>,
+        file: Option<PathBuf>,
+    },
+    Deduplicate,
+    Preprocess,
+}
+
+fn main() {
```
Comment (TheNewCivilian) [inline tools/regionalextracts/src/main.rs:45]: Maybe a general description would provide value here.

File: datatypes/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,4 @@
+pub mod persistence;
```
Comment (TheNewCivilian) [inline datatypes/src/lib.rs:1]: These are here as they are used by regionalextracts and the backend?

File: openlineplanner-backend/Cargo.toml (hunk 1 above)
Comment (zandemax) [reply openlineplanner-backend/Cargo.toml:15]: Ideally I want to see this merged into the geojson crate, I am currently trying to communicate with the maintainers if they are willing to merge that change.
I think manipulating the data on the FE side is worse than using this fork.

File: openlineplanner-backend/src/main.rs (hunk 2 above)
Comment (zandemax) [reply openlineplanner-backend/src/main.rs]: I think we can just straight up get rid of CORS except for local development

File: openlineplanner/src/components/MapStationPopup.vue (hunk 4 above)
Comment (zandemax) [reply openlineplanner/src/components/MapStationPopup.vue]: Yeah that did nothing, i will revert this

File: datatypes/src/lib.rs (hunk 8 above)
Comment (zandemax) [reply datatypes/src/lib.rs:1]: Yes, these are shared data types between the regionalextracts tool and the backend, so I extracted it into a seperate crate

File: openlineplanner-backend/src/main.rs (hunk 2 above)
Comment (TheNewCivilian) [reply openlineplanner-backend/src/main.rs]: True! We are now serving from the same origin right?

File: openlineplanner-backend/Cargo.toml (hunk 1 above)
Comment (TheNewCivilian) [reply openlineplanner-backend/Cargo.toml:15]: Why not put the same id here as outside? Wouldnt that fix it?

File: openlineplanner/src/components/MapStationPopup.vue (hunk 4 above)
Comment (TheNewCivilian) [reply openlineplanner/src/components/MapStationPopup.vue]: If thats extra work.. no need to. It will be changed anyway I guess..

File: openlineplanner-backend/Cargo.toml (hunk 1 above)
Comment (zandemax) [reply openlineplanner-backend/Cargo.toml:15]: Will be fixed in OLP-45

File: openlineplanner/src/components/MapStationPopup.vue (hunk 4 above)
Comment (zandemax) [reply openlineplanner/src/components/MapStationPopup.vue]: Done

**Review by TheNewCivilian — APPROVED** [review]
